ui/org.eclipse.pde.launching/src/org/eclipse/pde/launching/AbstractPDELaunchConfiguration.java

```diff
@@ -155,21 +155,24 @@ private String[] updateVMArgumentWithAdditionalArguments(String[] args, boolean
 			addModuleSystem = true;
 			argLength++; // Need to add the argument
 		}
-		IVMInstall vmInstall;
-		try {
-			vmInstall = VMHelper.getVMInstall(configuration);
-			if (vmInstall instanceof AbstractVMInstall) {
-				AbstractVMInstall install = (AbstractVMInstall) vmInstall;
-				String vmver = install.getJavaVersion();
-				if (vmver != null && JavaCore.compareJavaVersions(vmver, JavaCore.VERSION_17) >= 0) {
-					if (!argumentContainsAttribute(args, allowSecurityManager)) {
-						addAllowSecurityManager = true;
-						argLength++; // Need to add the argument
+
+		if (isEclipseBundleGreaterThanVersion(4, 24)) { // Don't add allow flags for eclipse before 4.24
```

Comment:
The int-arguments are not used. Furthermore I suggest to rename this method to something like `isSecurityManagerAllowSupported()` and to make the local variable `Version comparedVersion = new Version(4, 24, 0);` a static constant with an expressive name like `SECURITY_MANAGER_ALLOW_MIN_VERSION`.

Reply:
I have updated the PR to use incoming arguments and avoid List method.
Method name wise I feel this is good enough and there may be other checks in future.